### PR TITLE
chore: 🤖 migrate tslint setup to eslint.

### PR DESCRIPTION
**Pull request by corwin-of-amber**
It seems that `tslint` is deprecated.

https://github.com/palantir/tslint/issues/4534

I have done a "best effort" to create a compatible `eslint` configuration, utilizing `typescript-eslint`, which is supposed to be `tslint`'s direct successor.

There are more disabled rules now, because they were emitting errors on the current code base. Some you may wish to resolve in the code base, then re-enable.

Cheers 👋

### CODE REVIEW

**Comment by corwin-of-amber** [comment]
Should I rebase this?

**Comment by corwin-of-amber** [comment]
I see work on memfs has renewed lately. I am 100% willing to work on this fix, as well as [this one](https://github.com/corwin-of-amber/memfs/commit/8c4fa0f4f0a7583a3e84e6124d42ee396cf4b088) that allows using `buffer` version 6 (which I'm using so I need this). Just signal if you're available.

**Comment by G-Rath** [comment]
@corwin-of-amber sorry for the late response - I'm interested in landing this, if you wouldn't mind rebasing :)

File: eslint.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
```
Comment (G-Rath) [inline eslint.json:1]: Let's stick this in `.eslintrc.js` as that's more conventional

**Review by G-Rath — CHANGES_REQUESTED** [review]

**Pull request closed without merging**